content/testable-architecture.md

```diff
@@ -3,4 +3,62 @@ title: Testable Architecture
 description: Designing for testability
 ---
 
-Testable Architecture
\ No newline at end of file
+Automated tests and executable specifications can only work with software that has a testable architecture.
+A testable architecture enables fast and reliable tests that are easy to write, execute and maintain.
+Fast tests enable the developers to run them and obtain fast feedback on what they are building, without loosing focus or flow.
+
+When designing for testability, make sure that your products and services are composed of loosely-coupled,
+well-encapsulated components or modules; following the principles of
+[SOLID design](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)).
+
+Decoupling your business logic from your infrastructure (the often slow and brittle components),
+makes it possible to test at different levels, with maximum confidence and minimum cost.
+
+* Ensure that you can run your tests underneath your UI. Do not depend solely on UI tests. They are slow, expensive, and hard to fix.
+* Running tests against a database makes tests slow. And to make the tests behave consistently each time we run them,
+we need to make sure the database is in the expected state for each test, so the tests don't interfere with one another.
+
+We want our tests to use some kind of in-memory stub implementation, instead of an actual database.
+From the perspective of the domain logic, the behavior looks exactly the same.
+To make sure that the stub works in the same way as the real thing, we need to have confidence, which we can obtain using ‘contract tests’.
```

Comment:
Is the ornamentation around contract tests exactly what you want, here?

Reply:
Not sure what you mean here. Do you mean the ' ' around 'contract tests'? Copied that from the https://sdjournal.org/bdd-like-jazz/ blogpost.

Reply:
I mean the single quotes, yes.

Is "contract tests" not official naming, then perhaps single quotes are fine, but if we wish to make it stand out, then bold type could perhaps be helpful.

Reply:
See, I'm all weasel words and equivocating. Sorry about that.

Reply:
No problem. Thank you for proof reading all the time! It's hard to see the details when you've written and read the same doc several times. I *really, really* appreciate you taking the time to review; it makes the end result that much better!

Reply:
Not sure if 'contract tests' is the 'official' name (see, doing it again with the ' '). There seems to be a lot of misunderstanding about different types of tests all the time. I might just rewrite a little bit and avoid the term 'contract tests' (unless someone can provide a link/definition).